File: atcoder.jp/contests/abc153/tasks/abc153_b/main.ts
import { readFileSync } from "fs";
let inputLinesIndex = 0;
const inputLines = readFileSync("/dev/stdin", "utf8").split("\n");
const input = () => inputLines[inputLinesIndex++];

const sum = (array: number[]) => array.reduce((a, b) => a + b, 0);
const yesNo = (b: boolean | number) => (b ? "Yes" : "No");

const inputNumberArray = () => input().split(" ").map(Number);

const main = () => {
  const [H, N] = inputNumberArray();
  const A = inputNumberArray();

  const ans = H <= sum(A);
  console.log(yesNo(ans));
};

main();

File: atcoder.jp/contests/abc153/tasks/abc153_b/main.py
from typing import Union


def yes_no(b: Union[bool, int]) -> str:
    return "Yes" if b else "No"


input_int_list = lambda: list(map(int, input().split()))


H, N = input_int_list()
A = input_int_list()

ans = H <= sum(A)
print(yes_no(ans))
